private/bufpkg/bufplugin/plugin_full_name.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufplugin
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/bufbuild/buf/private/pkg/netext"
+)
+
+// PluginFullName represents the full name of the Plugin, including its registry, owner, and name.
+type PluginFullName interface {
+	// String returns "registry/owner/name".
+	fmt.Stringer
+
+	// Registry returns the hostname of the BSR instance that this Plugin is contained within.
+	Registry() string
+	// Owner returns the name of the user or organization that owns this Plugin.
+	Owner() string
+	// Name returns the name of the Plugin.
+	Name() string
+
+	isPluginFullName()
+}
+
+// NewPluginFullName returns a new PluginFullName for the given components.
+func NewPluginFullName(
+	registry string,
+	owner string,
+	name string,
+) (PluginFullName, error) {
+	return newPluginFullName(registry, owner, name)
+}
+
+// ParsePluginFullName parses a PluginFullName from a string in the form "registry/owner/name".
+func ParsePluginFullName(pluginFullNameString string) (PluginFullName, error) {
```

Comment:
Assume this would be pulled out to a `ParseFullName` for example

Reply:
This has now been pulled out into `bufparse.ParseFullName` and returns a type of `bufparse.FullName`. Updated the PR docs.